mica/starcheck/starcheck.py

```diff
@@ -200,12 +204,19 @@ def get_starcheck_catalog_at_date(date, starcheck_db=None):
     # are labeled by the planned obsid used in the starcheck output.  For observations
     # after SCS107 with SOSA, the kadi observation obsids and starcheck obsids don't match.
     # Use a helper function to line these up by starcat time.
-    obsid = get_starcheck_db_obsid(obs["starcat_date"], mp_dir)
+    if obs.get("starcat_date") is None:
+        # Observation has no star catalog (e.g. gyro hold) so use the kadi obsid. This
+        # may give an unexpected answer for a gyro hold non-catalog following SCS-107.
+        obsid = obs["obsid"]
+    else:
+        obsid = get_starcheck_db_obsid(obs["starcat_date"], mp_dir)
 
     # Use the obsid and the known products directory to use the more generic
     # get_starcheck_catalog to fetch the right one from the database
     cat_info = get_starcheck_catalog(obsid, mp_dir=mp_dir, starcheck_db=starcheck_db)
     cat_info["status"] = "ran" if date < CxoTime().date else "approved"
+    cat_info["obs"]["obsid_as_run"] = obs["obsid"]
```

Comment:
I'm assuming that tossing in another key doesn't hurt anything, but I'm not adamant about this addition.

Reply:
I'm the wrong person to judge this dispassionately because you know I'm always "Yay! More Info or Provenance"

Reply:
As long as you aren't aware of anything obvious that will break then I'm all in.